### PR TITLE
CI: move arm64 build to azure #38617

### PR DESCRIPTION
**Pull request by fangchenli**
xref #38617

### CODE REVIEW

**Comment by fangchenli** [comment]
I'm not sure why azure won't allow me to run a script. It says permission denied.

File: ci/docker_ci.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -e
+
```
Comment (jreback) [inline ci/docker_ci.sh:2]: is this chmod 755 ?

Comment (jreback) [reply ci/docker_ci.sh:2]: e.g. see run_tests.sh for an example of the mode we need

Comment (fangchenli) [reply ci/docker_ci.sh:2]: This line is identical to our other scripts. Not sure why it doesn't work.

Comment (jreback) [reply ci/docker_ci.sh:2]: no it's the file itself i think
so chmod 755 ci/docker_.... and commit it

File: azure-pipelines.yml [hunk 2]
```diff
@@ -27,27 +27,28 @@ jobs:
...
-        /bin/bash -xc "cd pandas && \
-        /opt/python/cp37-cp37m/bin/python -m venv ~/virtualenvs/pandas-dev && \
-        . ~/virtualenvs/pandas-dev/bin/activate && \
-        python -m pip install --no-deps -U pip wheel setuptools && \
-        pip install cython numpy python-dateutil pytz pytest pytest-xdist hypothesis pytest-azurepipelines && \
-        python setup.py build_ext -q -j2 && \
-        python -m pip install --no-build-isolation -e . && \
-        pytest -m 'not slow and not network and not clipboard' pandas --junitxml=test-data.xml"
-      displayName: 'Run 32-bit manylinux2014 Docker Build / Tests'
+        chomod 755 ci/docker_ci.sh
```
Comment (jreback) [inline azure-pipelines.yml:45]: i meant change it in the repo itself (but u can see if this is the problem)

File: ci/docker_ci.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+fi
+
+docker pull quay.io/pypa/$IMAGE
+docker run -v $(pwd):/pandas quay.io/pypa/$IMAGE \
+/bin/bash -xc "cd pandas && \
+/opt/python/cp37-cp37m/bin/python -m venv ~/virtualenvs/pandas-dev && \
+. ~/virtualenvs/pandas-dev/bin/activate && \
+python -m pip install --no-deps -U pip wheel setuptools && \
+pip install cython numpy python-dateutil pytz pytest pytest-xdist hypothesis pytest-azurepipelines && \
+python setup.py build_ext -q -j2 && \
```
Comment (jreback) [inline ci/docker_ci.sh:15]: can u increase threads for compilation ?

File: ci/docker_ci.sh [hunk 4]
```diff
@@ -12,6 +12,6 @@ docker run -v $(pwd):/pandas quay.io/pypa/$IMAGE \
 . ~/virtualenvs/pandas-dev/bin/activate && \
 python -m pip install --no-deps -U pip wheel setuptools && \
 pip install cython numpy python-dateutil pytz pytest pytest-xdist hypothesis pytest-azurepipelines && \
-python setup.py build_ext -q -j2 && \
+python setup.py build_ext -q -j4 && \
```
Comment (jreback) [inline ci/docker_ci.sh:15]: likely need to increase the number of processors for pytest as well (but get to build first)

**Comment by jreback** [comment]
should investigate how many cores that this chip has. i think we need to make this *much* faster. hasn't even compiled after 30 min

**Comment by fangchenli** [comment]
> should investigate how many cores that this chip has. i think we need to make this _much_ faster. hasn't even compiled after 30 min
> 
> maybe look at numpy recipe for aarch64

I think It's this slow due to that we need to use qemu to emulate arm64 in the container. NumPy uses Travis for arm64 build. I wonder how they are going to deal with this Travis credit issue.

**Comment by jreback** [comment]
> > should investigate how many cores that this chip has. i think we need to make this _much_ faster. hasn't even compiled after 30 min
> > maybe look at numpy recipe for aarch64
> 
> I think It's this slow due to that we need to use qemu to emulate arm64 in the container. NumPy uses Travis for arm64 build. I wonder how they are going to deal with this Travis credit issue.

so we *can* use travis.com and if its only for a couple of builds its fine. though that build also was taking > 1 hr which seems crazy.

**Comment by jreback** [comment]
i guess we can just do this on the new travis.com

**Comment by fangchenli** [comment]
close for now.

**Pull request closed without merging**